safe_api.js

```diff
@@ -78,10 +133,41 @@ window.safeAPI = {};
 					.then(() => mdHandle));
 		}
 
+		getString(key) {
+			return this.then(_ => {
+				console.log('Fetching key from handle: ', key, this.mdHandle);
+				return safeMutableData.get(this.mdHandle, key);
+			})
+			.then((val) => {
+				return val.buf.toString();
+			});
+		}
+
 		free() {
-			this.handlePromise.then((mdHandle) => {
-				safeMutableData.free(mdHandle);
-				this.handlePromise = null;
+			this.then((mdHandle) => {
+				// Nothing to do if no handles are provided
+				if ((mdHandle === undefined) && (this.mdHandle === undefined)) {
```

Comment:
where do you set this.mdHandle for the mutable data object?

Reply:
For many of these that I wasn't personally using, I just left the mdHandle as-is and only changed the `then()` calls. I was assuming that you already had your mdHandle passed properly in any code you already wrote.